jdaviz/utils.py

```diff
@@ -342,6 +344,85 @@ def indirect_units():
     return units
 
 
+def flux_conversion_general(values, original_unit, target_unit,
+                            equivalencies=None, with_unit=True):
+    """
+    This function converts `values` from `original_unit` to `target_unit` using the
+    provided `equivalencies` and accounting for special cases where unit converison is
+    not directly possible (e.g when u.spectral_density is needed and units are in surface
+    brightness per square pixel). Unit conversions in plugins should be passed through
+    this function rather than using astropy's unit.to() because this function handles
+    the special cases that might be encountered.
+
+    Note: This is essentially a simplified version of the function utils.flux_conversion.
+    The difference is that all required equivalencies must be passed in (rather than being
+    generated from an input spectrum or slice value, and combined with input equivalences).
+    I didn't want to replace calls to that function entirely because there is some extra
+    logic there that i'm unsure of, but in theory we could/should.
```

Comment:
Should this be rendered in API docstring? Looks like a note meant for internal devs only. Should this be moved to code comment instead?

Since this sounds like such an important function, please also document Parameters and Returns as laid out by numpydoc.